File: HCDInterviewCoach/DesignSystem/Spacing.swift
import SwiftUI

/// Spacing tokens with responsive values for mobile screens.
/// On macOS, standard desktop spacing is used.
/// On iPhone, larger spacing values are reduced for screen economy.
enum Spacing {
    static let xs: CGFloat = 4
    static let sm: CGFloat = 8
    static let md: CGFloat = 12
    static let lg: CGFloat = 16

    #if os(iOS)
    static var xl: CGFloat {
        UIDevice.current.userInterfaceIdiom == .phone ? 20 : 24
    }
    static var xxl: CGFloat {
        UIDevice.current.userInterfaceIdiom == .phone ? 32 : 40
    }
    #else
    static let xl: CGFloat = 24
    static let xxl: CGFloat = 40
    #endif

    /// Minimum recommended touch target size (44pt per HIG)
    static let touchTarget: CGFloat = 44
}

File: hcd-web/packages/ui/src/tokens/spacing.ts
// Spacing tokens ported from Swift Spacing.swift

export const spacing = {
  xxs: '0.125rem',  // 2px
  xs: '0.25rem',    // 4px
  sm: '0.5rem',     // 8px
  md: '0.75rem',    // 12px
  lg: '1rem',       // 16px
  xl: '1.5rem',     // 24px
  xxl: '2.5rem',    // 40px
} as const;

export const cornerRadius = {
  small: '0.25rem',   // 4px
  medium: '0.5rem',   // 8px
  large: '0.75rem',   // 12px
  xl: '1rem',         // 16px
  pill: '9999px',
} as const;

// Touch target minimum (44px for WCAG)
export const touchTarget = '2.75rem';
